src/formatDate.js

```diff
@@ -50,7 +50,54 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+
+  const fromArr = date.split(oldSeparator);
+  const lastIndex = toFormat.length - 1;
+
+  const oldDay = fromArr[fromFormat.findIndex((item) => item.includes('D'))];
+  const oldMonth = fromArr[fromFormat.findIndex((item) => item.includes('M'))];
+  const oldYear = fromArr[fromFormat.findIndex((item) => item.includes('Y'))];
+
+  const result = [];
+
+  for (let i = 0; i < lastIndex; i++) {
+    let current = '';
+
+    switch (toFormat[i]) {
+      case 'DD':
+        current = oldDay;
```

Comment:
```suggestion
        result.push(oldDay);
```
Can be pushed immediately